extension/src/popup/components/sendPayment/SendAmount/hooks/useSimulateTxData.tsx

```diff
@@ -380,18 +387,60 @@ function useSimulateTxData({
         Number("decimals" in assetBalance ? assetBalance.decimals : 7),
       );
 
+      // For Soroban transfers, check if contract supports muxed and determine final destination
+      let finalDestination = destination;
+      let sorobanMemo = memo;
+      if (simParams.type === "soroban") {
+        try {
+          const contractSupportsMuxed = await checkContractMuxedSupport({
+            contractId: tokenAddress,
+            networkDetails,
+          });
+          finalDestination = determineMuxedDestination({
```

Comment:
I think this can be in the `else` statement because we don't need to determine the muxed destination unless `contractSupportsMuxed `, right?